toolkit/analysis_and_visualization/classification/assets/scripts/pagination.js

```diff
@@ -0,0 +1,1126 @@
+/* eslint-disable no-useless-escape */
+/* eslint-disable no-empty */
+/* eslint-disable eqeqeq */
+/*global define*/
+
+/*
+ * pagination.js 2.1.5
+ * A jQuery plugin to provide simple yet fully customisable pagination.
+ * https://github.com/superRaytin/paginationjs
+ *
+ * Homepage: http://pagination.js.org
+ *
+ * Copyright 2014-2100, superRaytin
+ * Released under the MIT license.
+ */
+
+(function (global, $) {
+
+    if (typeof $ === "undefined") {
+        throwError("Pagination requires jQuery.");
+    }
+
+    var pluginName = "pagination";
+
+    var pluginHookMethod = "addHook";
+
+    var eventPrefix = "__pagination-";
+
+    // Conflict, use backup
+    if ($.fn.pagination) {
+        pluginName = "pagination2";
+    }
+
+    $.fn[pluginName] = function (options) {
+
+        if (typeof options === "undefined") {
+            return this;
+        }
+
+        var container = $(this);
+
+        var attributes = $.extend({}, $.fn[pluginName].defaults, options);
+
+        var pagination = {
+
+            initialize: function () {
+                var self = this;
+
+                // Cache attributes of current instance
+                if (!container.data("pagination")) {
+                    container.data("pagination", {});
+                }
+
+                if (self.callHook("beforeInit") === false) return;
+
+                // Pagination has been initialized, destroy it
+                if (container.data("pagination").initialized) {
+                    $(".paginationjs", container).remove();
+                }
+
+                // Whether to disable Pagination at the initialization
+                self.disabled = !!attributes.disabled;
+
+                // Model will be passed to the callback function
+                var model = self.model = {
+                    pageRange: attributes.pageRange,
+                    pageSize: attributes.pageSize
+                };
+
+                // dataSource`s type is unknown, parse it to find true data
+                self.parseDataSource(attributes.dataSource, function (dataSource) {
+
+                    // Currently in asynchronous mode
+                    self.isAsync = Helpers.isString(dataSource);
+                    if (Helpers.isArray(dataSource)) {
```

Comment:
Codacy found an issue: ['Helpers' was used before it was defined.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)